### PR TITLE
expression:rewrite password using new expression evaluation architecture

### PR DESCRIPTION
**Pull request by louishust**

### CODE REVIEW

**Comment by louishust** [comment]
@XuHuaiyu anyone review?

File: expression/builtin_encryption.go [hunk 1]
```diff
@@ -385,40 +386,38 @@ type passwordFunctionClass struct {
...
+	sig := &builtinPasswordSig{baseStringBuiltinFunc{bf}}
 	return sig.setSelf(sig), errors.Trace(c.verifyArgs(args))
 }
 
 type builtinPasswordSig struct {
-	baseBuiltinFunc
+	baseStringBuiltinFunc
 }
 
 // eval evals a builtinPasswordSig.
```
Comment (hanfei1991) [inline expression/builtin_encryption.go]: update this comment

File: expression/builtin_encryption_test.go [hunk 2]
```diff
@@ -302,22 +302,40 @@ func (s *testEvaluatorSuite) TestUncompressLength(c *C) {
...
 
-	fc := funcs[ast.PasswordFunc]
-	for _, test := range tests {
-		arg := types.NewDatum(test.in)
-		f, err := fc.getFunction(datumsToConstants([]types.Datum{arg}), s.ctx)
-		c.Assert(err, IsNil)
-		out, err := f.eval(nil)
+	fcName := ast.PasswordFunc
+	for _, t := range cases {
+		f, err := newFunctionForTest(s.ctx, fcName, primitiveValsToConstants(t.args)...)
```
Comment (XuHuaiyu) [inline expression/builtin_encryption_test.go]: sorry for late reply.
please add test for FieldType.

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**